src/ucp/rndv/rndv.c

```diff
@@ -1793,18 +1817,31 @@ UCS_PROFILE_FUNC(ucs_status_t, ucp_rndv_rtr_handler,
     ep_config = ucp_ep_config(ep);
     put_zcopy = &ep_config->rndv.put_zcopy;
 
-    ucs_assertv(rndv_rtr_hdr->sreq_id == sreq->send.msg_proto.sreq_id,
-                "received local sreq_id 0x%"PRIx64" is not equal to expected sreq_id"
-                " 0x%"PRIx64, rndv_rtr_hdr->sreq_id, sreq->send.msg_proto.sreq_id);
-    ucp_trace_req(sreq, "received rtr address 0x%"PRIx64" remote rreq_id"
-                  "0x%"PRIx64, rndv_rtr_hdr->address, rndv_rtr_hdr->rreq_id);
+    ucs_assertv(rndv_rtr_hdr->sreq_id == sreq->req_id.local,
+                "received local sreq_id 0x%" PRIx64 " is not equal to expected"
+                " sreq_id 0x%" PRIx64,
+                rndv_rtr_hdr->sreq_id, sreq->req_id.local);
+    ucp_trace_req(sreq,
+                  "received rtr address 0x%" PRIx64 " remote rreq_id "
+                  "0x%" PRIx64,
+                  rndv_rtr_hdr->address, rndv_rtr_hdr->rreq_id);
     UCS_PROFILE_REQUEST_EVENT(sreq, "rndv_rtr_recv", 0);
 
     if (sreq->flags & UCP_REQUEST_FLAG_OFFLOADED) {
         /* Do not deregister memory here, because am zcopy rndv may
          * need it registered (if am and tag is the same lane). */
         ucp_tag_offload_cancel_rndv(sreq);
         ucs_assert(!ucp_ep_use_indirect_id(ep));
+    } else if (rndv_rtr_hdr->offset == 0) {
+        /* Currently fragmented PUT doens't support tracking, since RTR handler
+         * untracks the send request for the first RTR message. If subsequent RTRs
+         * will be received when the PUT operation for the first RTR completed, it
```

Comment:
will be received -> did you mean will NOT be received?

Reply:
remove it at all